presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -475,8 +477,9 @@ private static boolean shouldUseFileSplitsFromInputFormat(InputFormat<?, ?> inpu
             }
             if (containsEligibleTableBucket) {
                 LocatedFileStatus file = files.get(partitionBucketNumber);
-                splitFactory.createInternalHiveSplit(file, readBucketNumber)
-                        .ifPresent(splitList::add);
+                eligibleTableBucketNumbers.stream()
+                        .map(tableBucketNumber -> splitFactory.createInternalHiveSplit(file, readBucketNumber, tableBucketNumber))
```

Comment:
Do we just want to do 
```java
splitFactory.createInternalHiveSplit(file, readBucketNumber, tableBucketNumber)
       .ifPresent(splitList::add);
```
?

IIRC if `tableBucketNumber > max(partitionBucketNumber, readBucketCount )`, we will hit this multiple times for the same Hive file-- so multiple splits will be created ?